pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/data/source/ColumnDataSource.java

```diff
@@ -61,12 +62,12 @@ public ColumnDataSource(ColumnIndexContainer indexContainer, ColumnMetadata meta
   public ColumnDataSource(FieldSpec fieldSpec, int numDocs, int maxNumMultiValues, DataFileReader forwardIndex,
       InvertedIndexReader invertedIndex, MutableDictionary dictionary) {
     this(fieldSpec.getName(), fieldSpec.getDataType(), fieldSpec.isSingleValueField(), false, numDocs,
-        maxNumMultiValues, forwardIndex, invertedIndex, dictionary);
+        maxNumMultiValues, forwardIndex, invertedIndex, dictionary, -1);
```

Comment:
Use static final variable instead of -1, for readability. Also, unclear from the context why -1 is being passed for cardinality instead of the actual value?

Reply:
I've updated the code to use UNKOWN_CARDINALITY for now mainly because it was not directly available for realtime segments. Will followup with Jackie on the different constructors for datasource and if/how we can get the cardinality in such cases.